### DISCUSSION TITLE
Version 2 - Updating - Clock or no clock

### DISCUSSION THREAD

**Discussion by fiatbravodriver** [sanni/cartreader#550] [Need help]
Hi guys

I am wanting to update my V2.

There is a note on the V2 wiki that says:

Updating the firmware:
This hardware variant runs the same Cart_Reader Arduino sketch as the others and can be updated to the newest version without any changes.

But on the instructions it states to remove the // on HW2 

Then to:

In case you do not have the Clock Generator installed you need to change the line "#define clockgen_installed" to "//#define clockgen_installed".

Is there an easy way to find out if my v2 has a clock generator. I cannot see a mention of it in the build guild so maybe was this for V3 and above?

Kind Regards

**Comment by sanni** [comment]
Turn your V2 around and look if you have this addon PCB:

![agegaagaggaga](https://user-images.githubusercontent.com/19169494/194716371-c993460d-26ef-4ae8-a424-b1763417f30f.jpg)

Then in Cart_Reader.ino change two things:

### 1)

If the addon PCB is installed on your Cart Reader you need:
````
#if (defined(HW2) || defined(HW3))
#define enable_OLED
#define enable_Button2
#define clockgen_installed
#define CA_LED
//#define fastcrc
#endif
````

If not you need:
````
#if (defined(HW2) || defined(HW3))
#define enable_OLED
#define enable_Button2
//#define clockgen_installed
#define CA_LED
//#define fastcrc
#endif
````

### 2)

In both cases you need to change this:

````
//******************************************
// !!! CHOOSE HARDWARE VERSION !!!
//******************************************
// Remove // in front of the line with your hardware version
// #define HW5
// #define HW4
// #define HW3
// #define HW2
// #define HW1
// #define SERIAL_MONITOR
````

into this:

````
//******************************************
// !!! CHOOSE HARDWARE VERSION !!!
//******************************************
// Remove // in front of the line with your hardware version
// #define HW5
// #define HW4
// #define HW3
#define HW2
// #define HW1
// #define SERIAL_MONITOR
````

**Reply by fiatbravodriver** [reply]
Hi Sani hope you are well. Sorry for they delay. Here is a photo. 

![IMG_0328](https://user-images.githubusercontent.com/43493640/208715955-b0c4e3e1-531a-4c9f-b197-c8ef2f0541f2.jpg)

Would this be the additional board?

Today I am trying to write back an n64.eep game save but it fails on Verifying. I am going to upgrade to the latest version to check. 

I also noticed that the Wiki does not have any instructions for writing a game save back. I imagine the switches are all set to down. 

Is that correct?

Many thanks and Seasons Greetings.

**Comment by sanni** [comment]
You don't have the clock generator so you need:

````
#if (defined(HW2) || defined(HW3))
#define enable_OLED
#define enable_Button2
//#define clockgen_installed
#define CA_LED
//#define fastcrc
#endif
````
